src/gui/entry/EditEntryWidget.cpp

```diff
@@ -994,7 +994,9 @@ bool EditEntryWidget::hasBeenModified() const
 
 void EditEntryWidget::togglePasswordGeneratorButton(bool checked)
 {
-    m_mainUi->passwordGenerator->regeneratePassword();
+    if (checked) {
+        m_mainUi->passwordGenerator->regeneratePassword();
```

Comment:
There is never any point in generating a new password if we are _hiding_ the password generator widget.